troubleshoot/troubleshoot.sh

```diff
@@ -0,0 +1,402 @@
+#!/bin/bash
+
+set -e
+
+selected_dynakube="dynakube"
+selected_namespace="dynatrace"
+cli="kubectl"
+default_oneagent_image="docker.io/dynatrace/oneagent"
+
+missing_value="<no value>"
+api_url=""
+paas_token=""
+log_section=""
+
+while [ $# -gt 0 ]; do
+  case "$1" in
+  --dynakube)
+    selected_dynakube="$2"
+    shift 2
+    ;;
+  --namespace)
+    selected_namespace="$2"
+    shift 2
+    ;;
+  --oc)
+    cli="oc"
+    shift 1
+    ;;
+  --openshift)
+    default_oneagent_image="registry.connect.redhat.com/dynatrace/oneagent"
+    shift 1
+    ;;
+  *)
+    echo "ERROR: unsupported option: '$1'"
+    exit 1
+    ;;
+  esac
+done
+
+log_info() {
+  printf "[%10s] %s\n" "$log_section" "$1"
+}
+
+error() {
+  printf "ERROR: %s\n" "$1"
+  exit 1
+}
+
+checkDependencies() {
+  if ! command -v jq &> /dev/null
+  then
+      error "jq is required to run this script!"
+  fi
+}
+
+checkNs() {
```

Comment:
You're not checking for Ns here ;-)
```suggestion
checkNamespace() {
```